### PR TITLE
[#158079564] added redis premium tier

### PR DESCRIPTION
**Pull request by gunzip**
we must set the connection string as k8 secret afterwards

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * ⚙️ [#158079564](https://www.pivotaltracker.com/story/show/158079564): Add the Redis resource through Terraform and put the address in the K8s configuration

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: infrastructure/azure.tf [hunk 1]
```diff
@@ -998,6 +1000,48 @@ resource "azurerm_dns_cname_record" "azurerm_dns_main_zone_dkim" {
...
+  name                      = "${local.azurerm_redis_backup_name}"
+  resource_group_name       = "${azurerm_resource_group.azurerm_resource_group.name}"
+  location                  = "${azurerm_resource_group.azurerm_resource_group.location}"
+  account_tier              = "Standard"
+  account_replication_type  = "GRS"
+  enable_blob_encryption    = true
+  enable_https_traffic_only = true
+}
+
+resource "azurerm_redis_cache" "azurerm_redis_cache" {
```
Comment (gunzip) [inline infrastructure/azure.tf]: same values atm for test and production, check these

File: infrastructure/azure.tf [hunk 2]
```diff
@@ -998,6 +1000,48 @@ resource "azurerm_dns_cname_record" "azurerm_dns_main_zone_dkim" {
...
 }
 
+#
+# Redis cache
+# 
+resource "azurerm_storage_account" "azurerm_redis_backup" {
+  name                      = "${local.azurerm_redis_backup_name}"
+  resource_group_name       = "${azurerm_resource_group.azurerm_resource_group.name}"
+  location                  = "${azurerm_resource_group.azurerm_resource_group.location}"
+  account_tier              = "Standard"
```
Comment (cloudify) [inline infrastructure/azure.tf]: pls link to the page that describes the tier

File: infrastructure/azure.tf [hunk 3]
```diff
@@ -998,6 +1000,48 @@ resource "azurerm_dns_cname_record" "azurerm_dns_main_zone_dkim" {
...
+  account_replication_type  = "GRS"
```
Comment (cloudify) [inline infrastructure/azure.tf]: pls link to the page that describes the replication type

File: infrastructure/azure.tf [hunk 4]
```diff
@@ -998,6 +1000,48 @@ resource "azurerm_dns_cname_record" "azurerm_dns_main_zone_dkim" {
...
+  enable_blob_encryption    = true
+  enable_https_traffic_only = true
+}
+
+resource "azurerm_redis_cache" "azurerm_redis_cache" {
+  name                = "${local.azurerm_redis_cache_name}"
+  location            = "${azurerm_resource_group.azurerm_resource_group.location}"
+  resource_group_name = "${azurerm_resource_group.azurerm_resource_group.name}"
+
+  # Possible values for Premium: 1, 2, 3, 4
```
Comment (cloudify) [inline infrastructure/azure.tf]: link to docs, what does it mean 1, 2, 3...?

Comment (cloudify) [reply infrastructure/azure.tf]: also, perhaps a short comment describing why we decided to use capacity=1 and shard_count=2

File: infrastructure/kubernetes/app-backend.yml [hunk 5]
```diff
@@ -51,16 +51,27 @@ spec:
         - name: "AZURE_NH_HUB_NAME"
           value: "agid-notificationhub-test"
         - name: "NODE_ENV"
-          value: "production"
+          # uses redis cluster when this is set to production
+          valueFrom:
+            configMapKeyRef:
+              name: app-backend-config
```
Comment (cloudify) [inline infrastructure/kubernetes/app-backend.yml]: we need a yml config for creating this config map (I actually don't see the real need to have a config map here?)

Comment (gunzip) [reply infrastructure/kubernetes/app-backend.yml]: The other option is to hardcode the value (ie. "agid-redis-test") and this configuration won't work in production.

This states for other configuration parameters as well (ie. the hardcoded "agid-notificationhub-test" value). I've openened a [pivotal story] https://www.pivotaltracker.com/story/show/158191461) to address the issue.

Other options to make k8 pods work in test *and* in production as well ?)

I guess this values should be taken from the Terraform state someway (ie. terraform show) and set up in the configuration map manually (using kubectl or by producing a yaml file). 

Note that there are already some secrets that are created manually (using kubectl) to make the APP backend work (and they don't appear in the configuration).

Comment (cloudify) [reply infrastructure/kubernetes/app-backend.yml]: it's ok for now to hardcode the config, we will have to create yml configs for production (or use helm templates) - we may have terraform create configuration maps in k8s, but I would not try to solve that problem now, unless it's trivial

Comment (gunzip) [reply infrastructure/kubernetes/app-backend.yml]: hardcoded

File: infrastructure/kubernetes/app-backend.yml [hunk 6]
```diff
@@ -58,7 +58,7 @@ spec:
               name: app-backend-secrets
               key: pre_shared_key
         - name: "REDIS_URL"
-          value: "io-development.redis.cache.windows.net"
+          value: "agid-redis-test.redis.cache.windows.net"
```
Comment (gunzip) [inline infrastructure/kubernetes/app-backend.yml]: cc @lussoluca changed hostname

**Pull request merged**